skxray/diff_roi_choice.py

```diff
@@ -0,0 +1,354 @@
+
+    # indices of rings
+    ring_inds = np.digitize(np.ravel(grid_values),
+                            np.array(q_r))
+    # discard the indices greater than number of rings
+    ring_inds[ring_inds > num_rings] = 0
+
+    # Edge values of each rings
+    ring_vals = []
+
+    for i in range(0, num_rings):
+        if i < num_rings:
+            ring_vals.append(q_r[i])
+            ring_vals.append(q_r[i + 1])
+        else:
+            ring_vals.append(q_r[num_rings-1])
+
+    ring_vals = np.asarray(ring_vals)
+
+    (ring_inds, ring_vals, num_pixels,
+     pixel_list) = _process_rings(num_rings, img_dim,
+                                  ring_vals, ring_inds)
+
+    return ring_inds, ring_vals, num_pixels, pixel_list
+
+
+def roi_rings_step(img_dim, calibrated_center, num_rings,
+                   first_r, delta_r, *args):
+    """
+    This will provide the indices of the required rings,
+    find the bin edges of the rings, and count the number
+    of pixels in each ring, and pixels list for the required
+    rings when there is a step value between rings.
+    Step value can be same or different steps between
+    each ring.
+
+    Parameters
+    ----------
+    img_dim: tuple
+        shape of the image (detector X and Y direction)
+        shape is [detector_size[0], detector_size[1]])
+
+    calibarted_center : tuple
+        defining the (x y) center of the image (mm)
+
+    num_rings : int
+        number of rings
+
+    first_r : float
+        radius of the first ring
+
+    delta_r : float
+        thickness of the ring
+
+    *args : tuple
+        step value for the next ring from the end of the previous
+        ring.
+        same step - same step values between rings (one value)
+        different steps - different step value between rings (provide
+        step value for each ring eg: 6 rings provide 5 step values)
+
+    Returns
+    -------
+    ring_vals : ndarray
+        edge values of the required rings
+
+    ring_inds : ndarray
+        indices of the required rings
+
+    num_pixels : ndarray
+        number of pixels in each ring
+
+    pixel_list : ndarray
+        pixel list for the required rings
+    """
+
+    grid_values = _grid_values(img_dim, calibrated_center)
+
+    ring_vals = []
+
+    for arg in args:
+        if arg < 0:
+            raise ValueError("step value for the next ring from the "
+                             "end of the previous ring has to be positive ")
+
+    if len(args) == 1:
+        #  when there is a same values of step between rings
+        #  the edge values of rings will be
+        ring_vals = first_r + np.r_[0, np.cumsum(np.tile([delta_r,
+                                                          float(args[0])],
+                                                         num_rings))][:-1]
+    else:
+        # when there is a different step values between each ring
+        #  edge values of the rings will be
+        if len(args) == (num_rings - 1):
```

Comment:
But I think checking step_value is better this way. Because if the len(args)==1 that is only for same value between rings. If I use elif here then I have to check separately if the user provided step value for each q ring.